### PR TITLE
Increase the test coverage of the Solution.

### LINKED ISSUE #15: v0.4.1 — Test Coverage Target (≥ 80%)
> *"Untested code is a liability disguised as a feature."*

#### The Problem Today

Test coverage is uneven across library projects. Some areas have thorough coverage; others — particularly edge cases in connector logic and validation — are undertested.

#### What We Are Building

A coverage measurement step in the CI pipeline (Coverlet + ReportGenerator) that fails the build if any library project falls below 80% line coverage. Gaps identified during this work are addressed before the 1.0.0 release.

#### Benefits

- A minimum coverage bar is enforced automatically on every pull request
- Regressions in test coverage are caught before they reach main

### PR DESCRIPTION
**Pull request by tsutomi**
This pull request introduces comprehensive unit tests for messaging-related exception and schema factory classes across several connectors (core, Facebook, Firebase, and Twilio). The new tests verify object construction, property assignments, interface implementations, and error handling for invalid arguments.

**Unit Test Additions:**

- **Messaging Exception Classes:**
  * Added unit tests for `MessagingException` to cover all constructors, property assignments, interface implementation, and argument validation.
  * Added unit tests for `MessageValidationException`, including tests for property assignments, interface implementation, and validation of error code arguments.
  * Added unit tests for `MessageValidationError` struct, verifying property assignments, null handling, interface implementation, and error code validation.

- **Channel Schema Factories:**
  * Added tests for `FacebookMessengerSchemaFactory` to ensure it returns the expected schema and matches static property values.
  * Added tests for `FirebasePushSchemaFactory` to verify schema creation and property values.
  * Added tests for both `TwilioSmsSchemaFactory` and `TwilioWhatsAppSchemaFactory`, checking correct schema instantiation and property matching.…uthenticationProvider, and messaging errors.

### CODE REVIEW

File: test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessagingExceptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        [InlineData("")]
+        [InlineData("   ")]
+        public void Should_ThrowArgumentNullException_When_ErrorCodeIsNullOrWhitespace(string? errorCode)
+        {
+            // Arrange
+            // Act & Assert
+            // ArgumentNullException.ThrowIfNullOrWhiteSpace throws ArgumentNullException for null,
+            // ArgumentException for empty/whitespace — both derive from ArgumentException
+            Assert.ThrowsAny<ArgumentException>(() => new MessagingException(errorCode!));
+        }
```
[the remark this answers is not in the document]
Comment (tsutomi) [reply test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessagingExceptionTests.cs:59]: @copilot apply changes based on [this feedback](https://github.com/deveel/deveel.messaging/pull/39#discussion_r3211223631)

File: test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessagingExceptionTests.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("   ")]
+        public void Should_ThrowArgumentNullException_When_ErrorCodeIsNullOrWhitespace_WithMessage(string? errorCode)
+        {
+            // Arrange
+            // Act & Assert
+            Assert.ThrowsAny<ArgumentException>(() => new MessagingException(errorCode!, "some message"));
+        }
```
[the remark this answers is not in the document]
Comment (tsutomi) [reply test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessagingExceptionTests.cs:104]: @copilot apply changes based on [this feedback](https://github.com/deveel/deveel.messaging/pull/39#discussion_r3211223648)

File: test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessagingExceptionTests.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("   ")]
+        public void Should_ThrowArgumentNullException_When_ErrorCodeIsNullOrWhitespace_WithInner(string? errorCode)
+        {
+            // Arrange
+            // Act & Assert
+            Assert.ThrowsAny<ArgumentException>(() => new MessagingException(errorCode!, "msg", null));
+        }
```
[the remark this answers is not in the document]
Comment (tsutomi) [reply test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessagingExceptionTests.cs:152]: @copilot apply changes based on [this feedback](https://github.com/deveel/deveel.messaging/pull/39#discussion_r3211223672)

File: test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessageValidationErrorTests.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        [InlineData("   ")]
+        public void Should_ThrowArgumentNullException_When_ErrorCodeIsNullOrWhitespace_TwoParam(string? errorCode)
+        {
+            // Arrange
+            var results = SampleResults();
+
+            // Act & Assert
+            Assert.ThrowsAny<ArgumentException>(() =>
+                new MessageValidationError(errorCode!, results));
+        }
```
[the remark this answers is not in the document]
Comment (tsutomi) [reply test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessageValidationErrorTests.cs:116]: @copilot apply changes based on [this feedback](https://github.com/deveel/deveel.messaging/pull/39#discussion_r3211223761)

File: test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessageValidationErrorTests.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        {
+            // Arrange
+            var results = SampleResults();
+
+            // Act & Assert
+            // ArgumentNullException.ThrowIfNullOrWhiteSpace → ArgumentNullException for null,
+            // ArgumentException for empty/whitespace (both extend ArgumentException)
+            Assert.ThrowsAny<ArgumentException>(() =>
+                new MessageValidationError(errorCode!, "message", results));
+        }
```
[the remark this answers is not in the document]
Comment (tsutomi) [reply test/Deveel.Messaging.Connector.Abstractions.XUnit/Unit/MessageValidationErrorTests.cs:73]: @copilot apply changes based on [this feedback](https://github.com/deveel/deveel.messaging/pull/39#discussion_r3211223740)
